genvm/vm.go

```diff
@@ -313,11 +313,11 @@ func (v *VM) execute(lctx ApplyContext, ss *core.StagedCache, txs []types.Transa
 		}
 		ctx := req.ctx
 		args := req.args
-		if ctx.PrincipalAccount.Balance < ctx.ParseOutput.FixedGas*ctx.ParseOutput.GasPrice {
-			logger.With().Warning("ineffective transaction. fixed gas not covered",
+		if intrinsic := core.ComputeIntrinsicGasCost(ctx.ParseOutput.BaseGas, tx.GetRaw().Raw, v.cfg.StorageCostFactor); ctx.PrincipalAccount.Balance < intrinsic {
```

Comment:
i am sure i asked this before.

why not check if `ctx.PrincipalAccount.Balance > ctx.Header.MaxGas` here since the first thing done in the execution flow is `ctx.Consume(ctx.Header.MaxGas)`?

Reply:
i think if transaction was included into the block we don't want to store it for free so we will try to get as much from it as possible. but if it can't cover this intrinsic gas (parse, verify and storage cost) and then we don't want to even try to execute it

Reply:
but this feels like we are coding for future when we have metered gas.
the code as is, doesn't make sense in the fixed gas model. consuming max gas is literally the first thing the code does.

Reply:
i don't understand what doesn't make sense.
code consumes max gas and if max gas is not available it will consume as much as it can

Reply:
ah. now i remembered. it will clean up the balance if max gas is not available. thanks